### DISCUSSION TITLE
use a record/object as a signal doesn't seem to work with react

### DISCUSSION THREAD

**Discussion by tcoopman** [preactjs/signals#487] [Q&A] [answered]
Before I create an issue, here is the reproduction: https://stackblitz.com/edit/react-signal-for-state-management-sr6tqk-vvyx3b?file=App.tsx

Bother elements use `useSignals()` but clicking doesn't trigger the other components. If I change the signal to be a number, everything seems to work fine.

**Comment by rschristian** [comment, marked as answer]
Don't use `Object.assign`, create a new object and assign that, i.e., `disclosure.value = {...disclosure.value, [id]: state }`

**Reply by tcoopman** [reply]
That works indeed, thanks!

**Reply by rschristian** [reply]
If you are looking for deep reactivity (objects), you may want to look into [`deepsignal`](https://github.com/luisherranz/deepsignal). It might be a bit more comfortable for these sorts of situations.
